src/open_inwoner/openzaak/import_export.py

```diff
@@ -0,0 +1,146 @@
+import dataclasses
+import io
+import json
+import logging
+from collections import defaultdict
+from typing import Any, Generator, Self
+
+from django.core import serializers
+from django.core.files.storage import Storage
+from django.db.models import QuerySet
+
+from .models import (
+    CatalogusConfig,
+    ZaakTypeConfig,
+    ZaakTypeInformatieObjectTypeConfig,
+    ZaakTypeResultaatTypeConfig,
+    ZaakTypeStatusTypeConfig,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class CatalogusConfigExport:
+    """Gather and export CatalogusConfig(s) and all associated relations."""
+
+    catalogus_configs: QuerySet
+    zaak_types: QuerySet
+    informatie_object_types: QuerySet
+    status_types: QuerySet
+    resultaat_types: QuerySet
+
+    def __iter__(self) -> Generator[QuerySet, Any, None]:
+        yield from (
+            self.catalogus_configs,
+            self.zaak_types,
+            self.informatie_object_types,
+            self.status_types,
+            self.resultaat_types,
+        )
+
+    def __eq__(self, other) -> bool:
+        for a, b in zip(self, other):
+            if a.difference(b).exists():
+                return False
+
+        return True
+
+    @classmethod
+    def from_catalogus_configs(cls, catalogus_configs: QuerySet) -> Self:
+        if not isinstance(catalogus_configs, QuerySet):
+            raise ValueError(
+                f"`catalogus_configs` is not a QuerySet, but a {type(catalogus_configs)}"
+            )
+
+        if catalogus_configs.model != CatalogusConfig:
+            raise ValueError(
+                f"`catalogus_configs` is of type {catalogus_configs.model}, not CatalogusConfig"
+            )
+
+        zaak_types = ZaakTypeConfig.objects.filter(catalogus__in=catalogus_configs)
```

Comment:
Are these `zaak_types` or `zaak_type_configs`? Same question below (and above in the `yield_from` at L34 ff.